src/functions-and-arrays.js

```diff
@@ -1,41 +1,98 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(a, b) {
+  return Math.max(a, b);
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(arr) {
+  let longestWord = []
+  if (arr.length === 0) {
+    return null}
+  else {
+  for (let i = 0; i < arr.length; i++) {    
+    if (longestWord.length === 0) {
+      longestWord.push(arr[i])
+    }
+    else if (arr[i].length > longestWord[0].length){
+      longestWord.splice(0,1,arr[i])
+    }
+    else continue;
+  }
+  return longestWord[0];
+}
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
+/* function sumNumbers(arr) {
+  let result = 0;
+  for (let i = 0; i < arr.length; i++){
+    result += arr[i];
+  }
+  return result;
+} */
+
+function sumNumbers(arr) {
+  let result = 0;
+  arr.forEach((value, index) => {
+    result += arr[index];
+  });
+  return result;
+}
 
 
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10, []];
 // Iteration #3.1 Bonus:
-function sum() {}
-
+function sum(arr) {
+  let result = 0;
+  for (let i = 0; i < arr.length; i++){
+    if (typeof arr[i] === 'string') {
+      result += arr[i].length;
+    } 
+    else if (typeof arr[i] === 'object' || typeof arr[i] === 'array'){
```

Comment:
change the order around - the last statement in the if/else chain should check errors, this way if you forgot to handle a datatype it will throw an error.